### ISSUE TITLE
Possible bug in prysm.thinfilms.multilayer_stack_rt

### ISSUE THREAD

**Issue by Jashcraf** [brandondube/prysm#92]
Hello! I'm currently working on a project to do some thin film optimization and wrote my own implementations of the BYU and Macleod multilayer matrix method for computing the effective fresnel coefficients. I was using prysm v0.21 as a reference and found a discrepancy in the calculations that I couldn't explain.

<img width="657" alt="image" src="https://user-images.githubusercontent.com/25557892/228886194-c76e616c-c3d5-4c1c-be7f-57a7c152aa18.png">

This is for a stack recipe shown below. It has the form `stack = [(n1,d1),(n2,d2),...]` where `n` is the refractive index and `d` is the film thickness in microns.
```
stack = [
            (f_n_SiO2(wvl),0.250),
            (f_n_BK7(wvl),0.100),
            (f_n_SiO2(wvl),0.250),
            (f_n_Al(wvl) + 1j*f_k_Al(wvl),0.0100),
            (f_n_BK7(wvl), 100_000_000)
        ]
```
where the `f_...` variables are `scipy.interpolate.CubicSpline()`'s of data from refractiveindex.info. 

I understand that there were substantive changes made to the thin film code during the "great vectorization" of release ~v0.20(?), so I tried computing the same data using the thin film code from v0.19. Turns out, the answers are much closer
<img width="657" alt="image" src="https://user-images.githubusercontent.com/25557892/228889039-22be27cc-5bac-4c7c-989d-7c7eeb23b003.png">

I don't think that this difference had been seen before because at 0 angle of incidence, the answers are the same between the two versions of prysm. 
<img width="657" alt="image" src="https://user-images.githubusercontent.com/25557892/228889965-4b4404d2-bb6d-459c-90ed-4616281a97e2.png">
The difference looks at the level of machine epsilon, so this discrepancy is indeed insidious
<img width="675" alt="image" src="https://user-images.githubusercontent.com/25557892/228890455-a00ba824-6742-4ef1-aecc-19c84bef2eaf.png">

The current tests in `test_thinfilm.py` all assume `aoi = 0`, so this wouldn't have been seen during the upgrade. When poking around I found in [line #501](https://github.com/brandondube/prysm/blob/0976ddb802be71772c30e0af1ad9d358bc1f7d31/prysm/thinfilm.py#L501) that the aor for the first thin film layer might be getting skipped.

Here we are concerned with a few variables, `indices, angles, thickness` are all 1D arrays that are the same length as the stack. 
`angles[0]` is the angle of incidence on the film
`angles[1]` is, if you assume an ambient vaccum, the angle of refraction from vacuum to the first layer, or the angle of the k-vector in the first layer.

Line 501 then loops from `i = (2,length(film_stack))`, and then populates `snell_aor(indices[i-1], indices[i], angles[i-1])`. Starting at `i = 2`, this sets `angle[2]` (the angle of light in the second film) equal to `snell_aor(indices[1],indices[2],angles[1])`, which is the angle of refraction between the 2nd and 3rd film layer, using the angle of incidence on the 2nd film layer.

Basically, I think the indices are offset from angles in the calculation. This wouldn't matter at normal incidence because all of the angles are equal to each other anyway.

I'm going to investigate further, but this seems like a (relatively) simple fix. The real hardship is making sure I don't break the generality of the code to broadcast the computation over different distances.

**Comment by Jashcraf** [comment]
Thinking about this more, I think it could replace `assume_vac_ambient`

**Comment by Jashcraf** [comment]
Made some changes and overall the code is now more simple at least - still working on the broadcasted calculations, but now poke.thinfilms, prysm.thinfilm, and some filmetrics data all agree

![image](https://github.com/brandondube/prysm/assets/25557892/80188c7b-b8e1-4892-b1ce-eda390285df4)

**Comment by Jashcraf** [comment]
^note that there is a slight discrepancy in the refractive indices we use and the ones filmetrics assumes

**Comment by Jashcraf** [comment]
The big "trick" to simplifying the code is just from snell's law. We know that
`n1*sin(theta1) == n2*sin(theta2)`
and
`n2*sin(theta2) == n3*sin(theta3)`

It follows then, that
`n1*sin(theta1) == n3*sin(theta3)`

The previous iteration of `multilayer_stack_rt` computes the angle of incidence _and_ angle of refraction for each layer, and had a case for assuming a non-vacuum ambient index, and for broadcasted computation. This lead to 4 iterations of code that was basically the same.

We cut it in half by computing the angle of refraction from the ambient index and angle of incidence!

**Comment by Jashcraf** [comment]
Tested the reflectance `Rs = np.abs(rs)**2` v.s. Poke for different angles of incidence and all looks well! It doesn't _yet_ broadcast over angle of incidence but perhaps this is a problem for another day

![image](https://github.com/brandondube/prysm/assets/25557892/d83401c1-a09c-4057-90df-ca13ec8cd89b)

**Comment by brandondube** [comment]
This was fixed in #99, thanks Jaren!
